gprofiler/utils/process.py

```diff
@@ -11,16 +11,22 @@
 from granulate_utils.linux.process import process_exe, read_proc_file
 from psutil import Process
 
+from gprofiler.platform import is_windows
+
 
 def search_proc_maps(process: Process, pattern: str) -> Optional[Match[str]]:
     return re.search(pattern, read_proc_file(process, "maps").decode(), re.MULTILINE)
 
 
 def process_comm(process: Process) -> str:
-    status = read_proc_file(process, "status").decode()
-    name_line = status.splitlines()[0]
-    assert name_line.startswith("Name:\t")
-    return name_line.split("\t", 1)[1]
+    if is_windows():
+        # TODO: Check if process is running
```

Comment:
What happens if the process is not running? Will it raise `NoSuchProcess`?

Reply:
Yes, a NoSuchProcess exception is raised. I included that comment because prior to some previous commits (from another author) there used to be a check to see if the process was still running.

Reply:
I see. In any case, raising `NoSuchProcess` is the intended behavior and that'll be handled by caller code. I suppose that could be documented. Can you remove this comment, and add a docstring to this function, something like
```
"""
:raises NoSuchProcess: if the process isn't running
"""
```

Reply:
Actually, process_comm does not raise an exception. However, the caller code raises CalledProcessError if the process is not running as follows:

`            except CalledProcessError as e:
                if (
                    b"Error: Failed to get process executable name. Check that the process is running.\n" in e.stderr
                    and not is_process_running(process)
                ):
                    logger.debug(f"Profiled process {process.pid} exited before py-spy could start")
                    return ProfileData(
                        self._profiling_error_stack("error", comm, "process exited before py-spy started"),
                        appid,
                        app_metadata,
                    )
                raise`

Reply:
I get `NoSuchProcess` if `process_comm` is called with a `Process` object that's already exited (on Linux). In any case, the comment was not needed